libafl/src/events/launcher.rs

```diff
@@ -542,27 +545,93 @@ impl<CF, MF, MT, S, SP> Debug for CentralizedLauncher<'_, CF, MF, MT, S, SP> {
         CoreId,
     ) -> Result<(), Error>,
     MF: FnOnce(
         Option<S>,
-        CentralizedEventManager<LlmpRestartingEventManager<(), S, SP>, SP>, // No hooks for centralized EM
+        CentralizedEventManager<StdCentralizedInnerMgr<S, SP>, SP>,
         CoreId,
     ) -> Result<(), Error>,
     MT: Monitor + Clone,
     S: State + HasExecutions,
     SP: ShMemProvider + 'static,
+{
+    /// Launch a standard Centralized-based fuzzer
+    pub fn launch(&mut self) -> Result<(), Error> {
+        let restarting_mgr_builder = |centralized_launcher: &Self, core_to_bind: CoreId| {
+            // Fuzzer client. keeps retrying the connection to broker till the broker starts
+            let builder = RestartingMgr::<(), MT, S, SP>::builder()
+                .always_interesting(centralized_launcher.always_interesting)
+                .shmem_provider(centralized_launcher.shmem_provider.clone())
+                .broker_port(centralized_launcher.broker_port)
+                .kind(ManagerKind::Client {
+                    cpu_core: Some(core_to_bind),
+                })
+                .configuration(centralized_launcher.configuration)
+                .serialize_state(centralized_launcher.serialize_state)
+                .hooks(tuple_list!());
+
+            #[cfg(feature = "adaptive_serialization")]
+            let builder = builder.time_ref(centralized_launcher.time_obs.handle());
+
+            builder.build().launch()
+        };
+
+        self.launch_generic(restarting_mgr_builder, restarting_mgr_builder)
+    }
+}
+
+#[cfg(all(unix, feature = "std", feature = "fork"))]
+impl<'a, CF, CIM, MF, MIM, MT, S, SP> CentralizedLauncher<'a, CF, CIM, MF, MIM, MT, S, SP>
+where
+    CF: FnOnce(Option<S>, CentralizedEventManager<CIM, SP>, CoreId) -> Result<(), Error>,
+    CIM: UsesState,
+    MF: FnOnce(
+        Option<S>,
+        CentralizedEventManager<MIM, SP>, // No hooks for centralized EM
+        CoreId,
+    ) -> Result<(), Error>,
+    MIM: UsesState,
+    MT: Monitor + Clone,
+    S: State + HasExecutions,
+    SP: ShMemProvider + 'static,
 {
     #[allow(clippy::similar_names)]
     #[allow(clippy::too_many_lines)]
-    /// launch the broker and the client and fuzz
-    pub fn launch(&mut self) -> Result<(), Error> {
+    /// Launch a Centralized-based fuzzer.
+    /// `main_inner_mgr_builder` will be called to build the inner manager of the main node.
+    /// `client_inner_mgr_builder` will be called to build the inner manager of the secondary nodes.
+    pub fn launch_generic<CIMF, MIMF>(
+        &mut self,
+        main_inner_mgr_builder: MIMF,
+        client_inner_mgr_builder: CIMF,
+    ) -> Result<(), Error>
+    where
+        CIMF: FnOnce(&Self, CoreId) -> Result<(Option<S>, CIM), Error>,
+        MIMF: FnOnce(&Self, CoreId) -> Result<(Option<S>, MIM), Error>,
+    {
+        let mut main_inner_mgr_builder = Some(main_inner_mgr_builder);
+        let mut client_inner_mgr_builder = Some(client_inner_mgr_builder);
```

Comment:
secondary_inner_mgr_builder?

i think main evaluator is `client` too,

Reply:
yes ur right its better

Reply:
i'll change other places where its not consistent with this notation